internal/mining/mining_harness_test.go

```diff
@@ -79,6 +79,54 @@ type fakeChain struct {
 	utxos                              *blockchain.UtxoViewpoint
 }
 
+// Returns a new fakeChain to be used for testing.
+// When using this function to create a fakeChain to pass to newMiningHarness(...), it is
+// recommended that the same chainParams used to call this function is used to call
+// newMiningHarness.
+// If the value of heightOfBestState is less than chainParams.CoinbaseMaturity, it will be
+// set to chainParams.CoinbaseMaturity
+func NewFakeChain(chainParams *chaincfg.Params, heightOfBestState int64) *fakeChain {
+	if heightOfBestState < int64(chainParams.CoinbaseMaturity) {
+		heightOfBestState = int64(chainParams.CoinbaseMaturity)
+	}
+
+	mockBestBlock := *dcrutil.NewBlock(&wire.MsgBlock{})
+	mockBestHash := mockBestBlock.Hash()
+	blocks := make(map[chainhash.Hash]*dcrutil.Block)
+	blocks[*mockBestHash] = &mockBestBlock
+
+	return &fakeChain{
+		blocks: blocks,
+		bestState: blockchain.BestState{
+			Hash:   *mockBestHash,
+			Height: heightOfBestState,
+		},
+		calcNextRequiredDifficulty:         chainParams.PowLimitBits,
+		calcNextRequiredDifficultyErr:      nil,
+		calcStakeVersionByHash:             0,
+		calcStakeVersionByHashErr:          nil,
+		checkConnectBlockTemplateErr:       nil,
+		checkTicketExhaustionErr:           nil,
+		checkTSpendHasVotesErr:             nil,
+		fetchUtxoEntryErr:                  nil,
+		fetchUtxoViewErr:                   nil,
+		fetchUtxoViewParentTemplateErr:     nil,
+		forceHeadReorganizationErr:         nil,
```

Comment:
You may just be doing this to be explicit, but most of these you are setting to the default value (`0`, `nil`, `false`, etc.), which is unnecessary.